src/test/e2e/20_zarf_init_test.go

```diff
@@ -27,13 +28,30 @@ func TestZarfInit(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.TODO(), 10*time.Minute)
 	defer cancel()
 
+	var (
+		mismatchedArch        = e2e.SetMismatchedArch()
+		initPackageVersion    = "UnknownVersion"
+		mismatchedInitPackage = fmt.Sprintf("zarf-init-%s-%s.tar.zst", mismatchedArch, initPackageVersion)
+	)
+
+	// Build init package with different arch than the cluster arch.
+	stdOut, stdErr, err := e2e.ExecZarfCommand("package", "create", ".", "--architecture", mismatchedArch, "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+	defer e2e.CleanFiles(mismatchedInitPackage)
+
+	// Check that `zarf init` fails in appliance mode when we try to initialize a k3s cluster
+	// on a machine with a different architecture than the package architecture.
+	// We need to use the --architecture flag here to force zarf to find the package.
+	_, stdErr, err = e2e.ExecZarfCommand("init", "--architecture", mismatchedArch, "--components=k3s", "--confirm")
+	require.Error(t, err, stdErr)
```

Comment:
It would be good to check the error message here as you did in the below test.

Reply:
Changes made